### PR TITLE
Add sample for dynamic settings

### PR DESCRIPTION
**Pull request by thomasyip-msft**
## Purpose

* Example on how to create dynamic list of settings, pass them to ARM Template and consume it in ExtensionType

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: samples/dynamiclabel-sample/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+page_type: sample
+languages:
+- yaml
+- json
+products:
+- azure-kubernetes-service
+- azure-marketplace
+---
+
+# Dynamic Label example
```
Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/README.md]: nit: Dynamic configuration settings

File: samples/dynamiclabel-sample/createUIDefinition.json [hunk 2]
```diff
@@ -0,0 +1,269 @@
+{
```
Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/createUIDefinition.json]: Can you please add CreateUIDefinition UX sample as well?

File: samples/dynamiclabel-sample/createUIDefinition.json [hunk 3]
```diff
@@ -0,0 +1,269 @@
...
+						"toolTip": "This will replace the value2's value inside your values.yaml",
+						"defaultValue": "Dogs",
+						"constraints": {
+							"required": true,
+							"regex": "^[a-z0-9A-Z ]{1,30}$",
+							"validationMessage": "Only alphanumeric characters are allowed, and the value must be 6-30 characters long."
+						}
+					},
+					{
+						"name": "labels",
```
Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/createUIDefinition.json]: Can we show a heading in the UX something like "Provide additional labels to your application pods. This is a optional setting."

File: samples/dynamiclabel-sample/mainTemplate.json [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+                "name": "inputs_array"
+            }
+        ],
+        "configurationSettings": "[if(or(variables('is_labels_defaultValue'), empty(parameters('labels'))), variables('azurevote-inputs'), union(variables('azurevote-inputs'), json(variables('inputs_t3'))))]",
+        "is_labels_defaultValue": "[equals(replace(string(parameters('labels')), ' ', ''), '[{\"labelName\":\"\",\"labelValue\":\"\"}]')]",
+        "inputs_string": "[concat(string(variables('inputs_array')), '_END_')]",
+        "inputs_t1": "[replace(variables('inputs_string'), '[{\"', '{\"')]",
+        "inputs_t2": "[replace(variables('inputs_t1'), '}]_END_', '}')]",
+        "inputs_t3": "[replace(variables('inputs_t2'), '},{\"', ',\"')]",
+        "azurevote-inputs": {
```
Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/mainTemplate.json]: nit: mandatory-inputs?

Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/mainTemplate.json]: Can we change the ordering of variables?

File: samples/dynamiclabel-sample/mainTemplate.json [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+        "clusterExtensionTypeName": "DONOTMODIFY",
+        "copy": [
+            {
+                "count": "[length(parameters('labels'))]",
+                "input": "[createObject(concat('labels.', parameters('labels')[copyIndex('inputs_array')].labelName), union(createObject('value', ''), parameters('labels')[copyIndex('inputs_array')]).labelValue)]",
+                "name": "inputs_array"
+            }
+        ],
+        "configurationSettings": "[if(or(variables('is_labels_defaultValue'), empty(parameters('labels'))), variables('azurevote-inputs'), union(variables('azurevote-inputs'), json(variables('inputs_t3'))))]",
+        "is_labels_defaultValue": "[equals(replace(string(parameters('labels')), ' ', ''), '[{\"labelName\":\"\",\"labelValue\":\"\"}]')]",
```
Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/mainTemplate.json]: nit: defaultState

Comment (RamyasreeChakka) [inline samples/dynamiclabel-sample/mainTemplate.json]: nit: change labels to dynamicSettings?

**Comment by RamyasreeChakka** [comment]
@atchutbarli - Can you please review this PR? It would be good if we can layout all the samples in a similar format/structure.

**Review by janezi7guo — APPROVED** [review]

**Review by RamyasreeChakka — APPROVED** [review]

**Pull request merged**